packages/react-ui-core/src/Gmap/OverlayView.js

```diff
@@ -0,0 +1,110 @@
+import React, { PureComponent } from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import classnames from 'classnames'
+
+@themed(/^Gmap_OverlayView/, { pure: true })
+export default class OverlayView extends PureComponent {
+  static propTypes = {
+    map: PropTypes.object,
+    anchor: PropTypes.object,
+    children: PropTypes.node,
+    className: PropTypes.string,
+    theme: PropTypes.object,
+  }
+
+  static defaultProps = {
+    className: '',
+    theme: {},
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.initOverlayView()
```

Comment:
This will probably fail on SSR since `window` isn't available in `initOverlayView`.

Reply:
Good point. I was thinking that because it's used as a child of `Gmap`, it will be guaranteed to render only in the browser. But one of the benefits of passing `map` as a prop rather than context is that `OverlayView` doesn't necessarily need to be a child of `Gmap` now.

Anyway, I moved that call to `componentDidMount` so this should be good now.

Reply:
Yeah, I was thinking of it being a child if it made a difference or not. I guess better to be safe than sorry 🤷‍♂️